core/src/test/scala/org/apache/spark/scheduler/SparkListenerSuite.scala

```diff
@@ -289,6 +290,17 @@ class SparkListenerSuite extends SparkFunSuite with LocalSparkContext with Match
     stageInfo.rddInfos.forall(_.numPartitions == 4) should be {true}
   }
 
+  test("SPARK-46383: Track TaskInfo objects") {
+    val conf = new SparkConf().set(DROP_TASK_INFO_ACCUMULABLES_ON_TASK_COMPLETION, true)
+    sc = new SparkContext("local", "SparkListenerSuite", conf)
+    val listener = new SaveActiveTaskInfos
+    sc.addSparkListener(listener)
+    val rdd1 = sc.parallelize(1 to 100, 4)
+    sc.runJob(rdd1, (items: Iterator[Int]) => items.size, Seq(0, 1))
+    sc.listenerBus.waitUntilEmpty()
+    listener.taskInfos.size should be { 0 }
```

Comment:
I am not sure I follow this test, what is it trying to do ?
This test will be successful even with `DROP_TASK_INFO_ACCUMULABLES_ON_TASK_COMPLETION` = `true`, right ? (Since it is simply checking for instance equality in the fired event ?)

Reply:
This test asserts that the same `TaskInfo` object is sent in the `onTaskStart` and `onTaskEnd` events. This test asserts the design in this PR that we are sending the original TaskInfo object to the DAGScheduler upon task completion and not a clone.

Reply:
Isn't that not simply an implementation detail ?
I am not sure what is the behavior we are testing for here - and how would this test help with some future change (and validation).

Reply:
I don't mind dropping it. I was just trying to assert one of the ways SparkListeners could be used. The test is more of a general test to ensure that we preserve the behavior of SparkListeners

Reply:
Functionally that (the right task info is in the event) should be covered already (in use of `SaveStageAndTaskInfo` for example). Do let me know if that is not the case.

Reply:
`SaveActiveTaskInfos` is caching `TaskInfos` but there are no tests on `TaskInfo` objects and none asserting that the `TaskInfo` objects are expected to remain the same across listener events